`ci-kubernetes-e2e-kind-rootless` failing before running tests, since 2025-10-15: `hack/lib/golang.sh: line 572: go_version[2]: unbound variable`

### Which jobs are failing?

ci-kubernetes-e2e-kind-rootless

### Which tests are failing?

Tests are not failing, but the compilation step is failing


https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-e2e-kind-rootless/1978364898429636608

```
go: go: could not create module cache: mkdir /go/src/k8s.io/kubernetes/_output: permission denied
mkdir: cannot create directory '/go/src/k8s.io/kubernetes/_output': Permission denied
mkdir: cannot create directory '/go/src/k8s.io/kubernetes/_output': Permission denied
go: go: could not create module cache: mkdir /go/src/k8s.io/kubernetes/_output: permission denied
/go/src/k8s.io/kubernetes/hack/lib/golang.sh: line 572: go_version[2]: unbound variable
/go/src/k8s.io/kubernetes/hack/lib/golang.sh: line 572: go_version[2]: unbound variable
make: *** [Makefile:97: all] Error 1
!!! [1015 07:49:21] Call tree:
!!! [1015 07:49:21]  1: build/../build/common.sh:405 kube::build::run_build_command_ex(...)
!!! [1015 07:49:21]  2: build/release-images.sh:39 kube::build::run_build_command(...)
make: Leaving directory '/home/rootless/go/src/k8s.io/kubernetes'
make: *** [Makefile:437: quick-release-images] Error 1
Failed to build Kubernetes: failed to build images: command "make -C /home/rootless/go/src/k8s.io/kubernetes quick-release-images 'KUBE_EXTRA_WHAT=cmd/kubeadm cmd/kubectl cmd/kubelet' KUBE_VERBOSE=0 KUBE_BUILD_HYPERKUBE=n KUBE_BUILD_CONFORMANCE=n KUBE_BUILD_PLATFORMS=linux/amd64" failed with error: exit status 2
ERROR: error building node image: failed to build kubernetes: failed to build images: command "make -C /home/rootless/go/src/k8s.io/kubernetes quick-release-images 'KUBE_EXTRA_WHAT=cmd/kubeadm cmd/kubectl cmd/kubelet' KUBE_VERBOSE=0 KUBE_BUILD_HYPERKUBE=n KUBE_BUILD_CONFORMANCE=n KUBE_BUILD_PLATFORMS=linux/amd64" failed with error: exit status 2
```

### Since when has it been failing?

2025-10-15


### Testgrid link

https://testgrid.k8s.io/sig-testing-kind#kind-rootless

### Reason for failure (if possible)

Likely to be a regression in:
- https://github.com/kubernetes/kubernetes/pull/134510

### Anything else we need to know?

_No response_

### Relevant SIG(s)

/sig testing

--- a/build/common.sh
+++ b/build/common.sh
@@ -322,6 +322,10 @@ function kube::build::destroy_container() {
   "${DOCKER[@]}" rm -f -v "$1" >/dev/null 2>&1 || true
 }
 
+function kube::build::is_docker_rootless() {
+  "${DOCKER[@]}" info --format '{{json .SecurityOptions}}' | grep -q "name=rootless"
+}
+
 # ---------------------------------------------------------------------------
 # Building
 
@@ -367,10 +371,11 @@ function kube::build::run_build_command_ex() {
 
   local -a docker_run_opts=(
     "--name=${container_name}"
-    "--user=$(id -u):$(id -g)"
     "--hostname=${HOSTNAME}"
     "-e=GOPROXY=${GOPROXY}"
   )
+
+  kube::build::is_docker_rootless || docker_run_opts+=("--user=$(id -u):$(id -g)")
 
   local detach=false
 

